cmd/vicadmin/usersession.go

```diff
@@ -69,31 +71,31 @@ func (u *UserSessionStore) Delete(id string) {
 	delete(u.sessions, id)
 }
 
-// Grabs the UserSession metadta object and doesn't establish a connection to vSphere
+// Grabs the UserSession metadata object and doesn't establish a connection to vSphere
 func (u *UserSessionStore) UserSession(id string) *UserSession {
 	u.mutex.RLock()
 	defer u.mutex.RUnlock()
 	return u.sessions[id]
 }
 
-// Get logs into vSphere and returns a vSphere session object. Caller responsible for error handling/logout
-func (u *UserSessionStore) VSphere(ctx context.Context, id string) (vSphereSession *session.Session, err error) {
+// Returns a vSphere session object. Caller responsible for error handling/logout
+func (u *UserSessionStore) VSphere(ctx context.Context, id string) (*session.Session, error) {
 	us := u.UserSession(id)
 	if us == nil {
 		return nil, fmt.Errorf("User session with unique ID %s does not exist", id)
 	}
-
-	if us.vsphere != nil {
-		log.Infof("Found cached vSphere session for vicadmin usersession %s", id)
-		return us.vsphere, nil
+	if us.vsphere == nil {
+		return nil, fmt.Errorf("No vSphere session found. User with ID %s must authenticate again.", id)
 	}
 
-	log.Infof("Creating vSphere session for vicadmin usersession %s", id)
-	s, err := vSphereSessionGet(us.config)
-	if err != nil {
-		return nil, err
-	}
-	us.vsphere = s
+	// log.Infof("Creating vSphere session for vicadmin usersession %s", id)
```

Comment:
This is where, last night, we were discussing what happens if vSphere expires the session or something like that -- the user just has to log in again. Perfectly OK behavior IMHO